packages/docs-reanimated/docs/shared-element-transitions/overview.mdx

```diff
@@ -40,90 +93,72 @@ To create a shared transition animation between two components on different scre
 
 ## Custom animation
 
-1. You need to create custom transition with Reanimated API, for example:
+You can create custom animation by using `SharedTransition.custom` for regular animations or `SharedTransition.progressAnimation` for progress-based animations.
 
 ```js
 import { SharedTransition } from 'react-native-reanimated';
+
 const transition = SharedTransition.custom((values) => {
   'worklet';
   return {
     height: withSpring(values.targetHeight),
     width: withSpring(values.targetWidth),
   };
-});
+})
+  .progressAnimation((values, progress) => {
+    'worklet';
+    const getValue = (
+      progress: number,
+      target: number,
+      current: number
+    ): number => {
+      return progress * (target - current) + current;
+    };
+    return {
+      width: getValue(progress, values.targetWidth, values.currentWidth),
+      height: getValue(progress, values.targetHeight, values.currentHeight),
+    };
+  })
+  .defaultTransitionType(SharedTransitionType.ANIMATION);
 ```
 
-2. You need to add custom transition as `sharedTransitionStyle` prop to your both components (on both screens):
+By combining these methods and setting `defaultTransitionType(SharedTransitionType.ANIMATION)`, you specify that the custom animation will be used for screen transitions, while the progress-based animation will be applied during swipe back gesture (**iOS only**).
+
+Setting `defaultTransitionType(SharedTransitionType.PROGRESS_ANIMATION)` will apply progress-based animation in both scenarios.
 
-```tsx {2,3}
+Then, to use it, simply pass your custom animation as `sharedTransitionStyle` prop to both your screen's components (just as with the `sharedTransitionTag`).
```

Comment:
I think it would be nice to inform users which component is responsible for which transition. Do they have to use the exact same transition on `sharedTransitionStyle`, or can they have different ones and the transitions will look differently depending on the component we are starting it from? I don't know it either so I think it would be good to specify if you find out 😄

Reply:
yes, good idea!! ✨